### PR TITLE
Blog App Milestone 11: Feat/add api endpoints

### PR DESCRIPTION
**Pull request by andres-condezo**
Rails

---
### Hi there :wave:
Thanks for taking the time to review my project.

--- 
### Changes

In this PR we:

- Created an API endpoint to list all posts for a user.
- Created an API endpoint to list all comments for a user's post.
- Created an API endpoint to add a comment to a post.
- Added authentication to your API endpoints.

---
### Get Started

Clone the repository :
```
git clone https://github.com/andres-condezo/Blog-app.git
cd Blog-app
git checkout feat/add_api_endpoints
```

### Any comment or suggestion will be thankfully received.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,7 +1,16 @@
 Rails.application.routes.draw do
+  post '/login', to: 'auth#login', default: {:format => :json}
```
Comment (DeliceLydia) [inline config/routes.rb:4]: - [x] Please add Registration endpoint

File: config/routes.rb [hunk 2]
```diff
@@ -1,7 +1,16 @@
 Rails.application.routes.draw do
+  post '/login', to: 'auth#login', default: {:format => :json} 
   # Defines the root path route ("/")
   root to: 'users#index'
 
+  scope 'api' do
+    get '/posts/:id/comments' => 'api/comments#index', as: :api_post_comments
```
Comment (DeliceLydia) [inline config/routes.rb]: - [x] Please add an API endpoint that returns all comments for a post: `/users/:user_id/posts/:id/comments`

File: config/routes.rb [hunk 3]
```diff
@@ -1,7 +1,16 @@
...
+    post '/posts/:id/comments' => 'api/comments#create', as: :api_post_comments_create
```
Comment (DeliceLydia) [inline config/routes.rb]: - [x] Please add an API endpoint that returns all comments for a post: `/users/:user_id/posts/:id/comments`

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there! Highlights are:
- [ ] Awesome README
- [ ] No linter errors

## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @andres-condezo 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉 

**Highlights**
- Great work making the required changes from the previous reviewer

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**